contracts/src/v0.8/functions/v1_3_0_zksync/libraries/ZKSyncChainSpecificUtil.sol

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {SYSTEM_CONTEXT, ISystemContext} from "../interfaces/zksync/ISystemContext.sol";
+
+/// @dev A library that abstracts out opcodes that behave differently across chains.
+/// @dev The methods below return values that are pertinent to the given chain.
+/// @dev The exisiting library used within other functions contracts contains opcodes not supported by ZKSync
+library ZKSyncChainSpecificUtil {
+  function _getCurrentTxL1GasFees() internal view returns (uint256 l1FeeWei) {
+    ISystemContext systemContext = ISystemContext(SYSTEM_CONTEXT);
+
+    // blob_gas_price_on_l1 * gas_per_byte ~= gas_price_on_l2 * l2_gas_PerPubdataByte
+    return systemContext.gasPrice() * systemContext.gasPerPubdataByte();
```

Comment:
i feel this is not doing what the func sig indicates?

what we have here is `L2 gas price * L2 gas per L1 pubdata / byte` so it still needs the number of bytes current tx will be updating, in order to correctly represent L1 gas cost in L2 world. however, this brings us back to the problem where it's impossible to know how many L1 bytes will be updated beforehand.

`L1 gas cost = `